### PR TITLE
fix: [CDS-82385]: Returning an Object instance instead of null to avoid exception

### PR DESCRIPTION
**Pull request by GoelPratyush99**
## Describe your changes
Earlier we were returning null delegate response due to which the delegate was not able to assign the PT (instance sync for Helm) task. Now instead of returning a null value we have created a DummyDelegateResposeData that will provide the non null and delegate can be assigned the task.
Screenshot 2023-10-30 at 11 20 35 AM

Screenshot 2023-10-30 at 11 18 14 AM

Screenshot 2023-10-30 at 11 18 29 AM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54745)

### CODE REVIEW

**Comment by GoelPratyush99** [comment]
trigger uts

**Comment by GoelPratyush99** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/DummyDelegateResponseData.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import io.harness.delegate.beans.DelegateResponseData;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@OwnedBy(HarnessTeam.CDP)
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = false, components = {HarnessModuleComponent.CDS_K8S})
+public class DummyDelegateResponseData implements DelegateResponseData {}
```
Comment (alebuh) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/DummyDelegateResponseData.java:24]: Do we really need a new class for this response? Could we use some existing DelegateResponse with empty values (for example DelegateStringResponseData)? I feel like it's not necessary to add one more class into kryo

Comment (GoelPratyush99) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/instancesync/DummyDelegateResponseData.java:24]: This class seem be developed earlier for a different purpose which is not in use anymore. Not sure if there would be any usecase for it anymore. This Dummy Delegate Response Data class will be exclusively for use of instance sync Container Validation Task (ContainerDummyTask.java) as it returned null which is not correct if we want the delegate to get assigned a task.

**Comment by GoelPratyush99** [comment]
trigger messagecheck

**Review by acasian-harness — APPROVED** [review]

**Review by markoharness — APPROVED** [review]

**Review by namantalaycha10 — APPROVED** [review]

**Review by abhinavsingh1196 — APPROVED** [review]

**Comment by GoelPratyush99** [comment]
trigger review

**Review by TarunUba1 — APPROVED** [review]

**Pull request merged**